superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -260,6 +264,33 @@ function nvd3Vis(element, props) {
     return types.indexOf(vizType) >= 0;
   }
 
+  function findYAxisField(metrics, publishedColumns) {
+    let columnName = '';
+
+    if (metrics && publishedColumns) {
+
+      publishedColumns.forEach((column) => {
```

Comment:
You can refactor this multiple forEach loop inside to each other more efficiently with .find .

Reply:
For an intersection of 2 arrays, according to different stackoverflow post a performant way of doing it would be with `find` and `includes`.

Reply:
<img width="754" alt="Screenshot 2019-05-31 at 12 08 30 AM" src="https://user-images.githubusercontent.com/18305698/58655848-b899b880-8338-11e9-9923-72a840a5d493.png">

Seems like a bug. Due to which I cannot check further the behavior of annotation layer Timeseries

Reply:
Even if the root of this issue is fixed, I think we should also implement a protection mechanism inside this method.